### PR TITLE
Add some SSH notes

### PR DESCRIPTION
**Pull request by vinokurig**
## What does this pull request change?
Add some info about starting a workspace from SSH Urls

## What issues does this pull request fix or reference?
https://github.com/eclipse-che/che-server/pull/539

## Specify the version of the product this pull request applies to
next
## Pull Request checklist

The author and the reviewers validate the content of this pull request with the following checklist, in addition to the [automated tests](code_review_checklist.adoc).

- Any procedure:
  - [ ] Successfully tested.
- Any page or link rename:
  - [ ] The page contains a redirection for the previous URL.
  - Propagate the URL change in:
    - [ ] Dashboard [default branding data](https://github.com/eclipse-che/che-dashboard/blob/main/packages/dashboard-frontend/src/services/bootstrap/branding.constant.ts)
- [ ] Builds on [Eclipse Che hosted by Red Hat](https://workspaces.openshift.com).
- [ ] the *`Validate language on files added or modified`* step reports no vale warnings.

### CODE REVIEW

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (ibuziuk) [inline modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: for private repos both PAT and OAuth should work, right?

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (amisevsk) [inline modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: This point should be stripped or reworked for downstreaming; I don't think we want to point to the upstream github repo for docs in downstream. 

Maybe we can adopt the `Configuring DevWorkspaces to use SSH keys for Git operations` section into Che docs? The file is already in `.adoc` format, though would need a style pass, etc.

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (vinokurig) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: yes

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (vinokurig) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: I don't think we need to have a dedicated page about configuring SSH by a secret. When https://github.com/eclipse/che/issues/22111 is done we will point to the user friendly dashboard procedure, so the current procedure will be redundant as a too complicated one. @ibuziuk **WDYT?**

Comment (amisevsk) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: I agree we don't need a dedicated page and the link to DWO's doc is sufficient here. My concern is that we need to do some additional work when we downstream these docs later, as I don't think we should be linking to Github there (this is not a necessarily concern for this repo, though).

**Comment by vinokurig** [comment]
@deerskindoll Could you please review the PR?

**Comment by deerskindoll** [comment]
I created a PR that will fix the issue with building and validating the PR but it needs to be approved before I can merge it: https://github.com/eclipse-che/che-docs/pull/2643

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (deerskindoll) [inline modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]:
```suggestion
With {prod-short}, you can use either an HTTPS or SSH URL to start a new workspace with a clone of a Git repository.
```

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (deerskindoll) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: if I understand it correctly, you'll make this part visible later?

Comment (deerskindoll) [inline modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]:
```suggestion
* If you use a SSH URL to start a new workspace, you must propagate the SSH key must be propagated. See link:https://github.com/devfile/devworkspace-operator/blob/main/docs/additional-configuration.adoc#configuring-devworkspaces-to-use-ssh-keys-for-git-operations[Configuring DevWorkspaces to use SSH keys for Git operations] for more information.
```

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (deerskindoll) [inline modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]:
```suggestion
* If the SSH URL points to a private repository, you must apply an access token to be able to fetch the `devfile.yaml` content. You can do this either by accepting an SCM authentication page or following a xref:using-a-git-provider-access-token.adoc[Personal Access Token] procedure.
```

**Review by deerskindoll — COMMENTED** [review]
I left a couple of suggestions, please check them out.

File: modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc
Comment (vinokurig) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: @deerskindoll 
> if I understand it correctly, you'll make this part visible later?

This is already visible:
screenshot-0 0 0 0_4000-2023 08 23-15_51_13
It should be modified later

Comment (deerskindoll) [reply modules/end-user-guide/pages/starting-a-new-workspace-with-a-clone-of-a-git-repository.adoc]: got it! thanks for the clarification

**Review by deerskindoll — COMMENTED** [review]
lgtm

**Comment by vinokurig** [comment]
@deerskindoll Could you please approve the PR for the `merge button` to be available.

**Review by deerskindoll — APPROVED** [review]
+

**Comment by deerskindoll** [comment]
> @deerskindoll Could you please approve the PR for the `merge button` to be available.

sorry, it was a misclick

**Pull request merged**
